src/network/clientpackethandler.cpp

```diff
@@ -370,27 +370,20 @@ void Client::handleCommand_ChatMessage(NetworkPacket *pkt)
 		wstring message
 	 */
 
-	ChatMessage *chatMessage = new ChatMessage();
+	ChatMessage msg;
 	u8 version, message_type;
 	*pkt >> version >> message_type;
 
-	if (version != 1 || message_type >= CHATMESSAGE_TYPE_MAX) {
-		delete chatMessage;
+	if (version != 1 || message_type >= CHATMESSAGE_TYPE_MAX)
 		return;
-	}
 
-	*pkt >> chatMessage->sender >> chatMessage->message >> chatMessage->timestamp;
+	*pkt >> msg.sender >> msg.text >> msg.timestamp;
 
-	chatMessage->type = (ChatMessageType) message_type;
+	msg.type = (ChatMessageType) message_type;
 
 	// @TODO send this to CSM using ChatMessage object
-	if (!moddingEnabled() || !m_script->on_receiving_message(
-			wide_to_utf8(chatMessage->message))) {
-		pushToChatQueue(chatMessage);
-	} else {
-		// Message was consumed by CSM and should not handled by client, destroying
-		delete chatMessage;
-	}
+	if (!moddingEnabled() || !m_script->on_receiving_message(msg))
+		pushToChatQueue(new ChatMessage(msg));
```

Comment:
why this copy ? to prevent a memleak handled in the previous case ??

Reply:
`msg` is destructed after this function call, thus copying is better in this case because
1) the chat queue takes care of the pointer
2) lower risk of forgetting to delete it in this function when it's not pushed to the queue